src/vue/vuex-modules/data/auth-module.js

```diff
@@ -230,6 +212,51 @@ AuthModule.api = (moduleInstance, store) => {
       })
     },
 
+    /**
+     * Sets a state of an authentication data to the one provided.
+     * @param {AuthData} authData - An object containing an authentication data.
+     */
+    updateAuthData: (authData) => {
+      moduleInstance._authData = authData
+
+      if (moduleInstance.config.queryParams && moduleInstance.config.queryParams.sessionDuration) {
+        const customSessionDuration = moduleInstance.config.queryParams.sessionDuration
+        // Do not allow to override session duration to values higher than set by the auth server
+        if (customSessionDuration && Number(customSessionDuration) * 1000 < moduleInstance._authData.expirationInterval) {
+          moduleInstance._authData.setSessionDuration(Number(customSessionDuration) * 1000)
+        }
+      }
+
+      store.commit('auth/setIsAuthenticated', moduleInstance._authData)
+
+      if (moduleInstance._authData.hasSessionExpired) {
+        moduleInstance._api.expireSession()
```

Comment:
should we also publish the SESSION_EXPIRED event here?

Reply:
ah, so it's really the opposite scenario of when we publish the event.  It might be worth adding a few comments to the code here to remind ourselves of the scenario. Thanks!

Reply:
>  It might be worth adding a few comments to the code here to remind ourselves of the scenario.

That's a great idea. I will add that.